core/src/main/java/org/apache/druid/math/expr/BinaryLogicalOperatorExpr.java

```diff
@@ -74,7 +74,7 @@ public ExprType getOutputType(InputBindingInspector inspector)
   @Override
   public boolean canVectorize(InputBindingInspector inspector)
   {
-    return inspector.areNumeric(left, right) && inspector.canVectorize(left, right);
```

Comment:
i _think_ canVectorize will still end up false because array types/functions are not yet vectorizable, but yeah i guess array types would not be well handled here even if they were, though maybe that is the case for non-vectorized expression as well. I'll do a bit of exploration and see what up

Reply:
Did this exploration lead somewhere?

Reply:
array types do not currently implement `canVectorize` so will return the default false. We probably should not use `canVectorize` for argument validation and maybe should consider doing this explicitly when we can resolve the input types I think.

Reply:
Got it. IMO we shouldn't rely on the fact that the array types happen to not be vectorizable. If they suddenly became vectorizable then this method would start returning true, maybe improperly.

Reply:
like, I agree here, but `canVectorize` still feels wrong to saddle with type validation, it can explode when creating the processor if the types are nonsense

I think what might be best is to introduce a 'validate' method to expressions so that arguments can be checked up front, and maybe explode if the input is madness

Reply:
In that case, could you add a comment about the situation?